bookshelf/bookshelf.d.ts

```diff
@@ -177,8 +191,12 @@ declare module 'bookshelf' {
 			keys() : string[];
 			last() : T;
 			lastIndexOf(value : any, fromIndex? : number) : number;
-			map(predicate? : Lodash.ListIterator<T, boolean>|Lodash.DictionaryIterator<T, boolean>|string, thisArg? : any) : T[];
+
+			// See https://github.com/Microsoft/TypeScript/blob/v1.8.10/lib/lib.core.es7.d.ts#L1122
```

Comment:
You are right referencing a new `U` parameter. I don't know why I referenced `boolean` in the initial implementation. My thought was that it might be confusing to reference this es7 method since the lodash method overrides it (even though the API is almost identical).